Text/Mustache/Compile.hs

```diff
@@ -99,10 +120,10 @@ compileMustacheText pname txt =
 
 -- | Check if a given 'FilePath' points to a mustache file.
 
-isMustacheFile :: FilePath -> IO Bool
-isMustacheFile path = do
+hasExtension :: FilePath -> String -> IO Bool
+hasExtension path ext = do
   exists <- doesFileExist path
-  let rightExtension = F.takeExtension path == ".mustache"
```

Comment:
I wonder if we could somehow allow to match on files without any extension at all. Maybe we could accept extensions with preceding dot in `compileMustacheDirCustom` then we could pass `""` there.

More generally, would it be possible to accept a predicate we could use to select templates instead of extension? If we are to generalize template selection, this seems to be the right thing to do.